### PR TITLE
remove tools/setup_helpers/cudnn.py

### PR DESCRIPTION
**Pull request by xuhdev**
FindCUDNN.cmake and cuda.cmake have done the detection. This commit deletes `tools/setup_helpers/cudnn.py` as it is no longer needed.

Previously in #25482, one test failed because TensorRT detects cuDNN differently, and there may be situations we can find cuDNN but TensorRT cannot. This is fixed by passing our detection result down to TensorRT.

### CODE REVIEW

**Comment by xuhdev** [comment]
caffe2_py2_cuda9_0_cudnn7_ubuntu16_04_build passes now: https://circleci.com/gh/pytorch/pytorch/2730080?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by xuhdev** [comment]
@ezyang The second commit is the new stuff compared with the previous version. The third commit is the CI tweaks. (Hopefully, the conflict will go away once the CI tweaking commit is reverted)

**Comment by ezyang** [comment]
> caffe2_py2_cuda9_0_cudnn7_ubuntu16_04_build passes now: https://circleci.com/gh/pytorch/pytorch/2730080?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

What's the tweak that fixed this one?

**Comment by xuhdev** [comment]
It's the second commit: https://github.com/pytorch/pytorch/pull/25876/commits/59a65e3bc514368f18c0c4e959671d6d71a54d31

File: tools/setup_helpers/cmake.py [hunk 1]
```diff
@@ -217,6 +217,7 @@ def generate(self, version, cmake_python_library, build_python, build_test, my_e
             '_GLIBCXX_USE_CXX11_ABI': 'GLIBCXX_USE_CXX11_ABI',
             'USE_CUDA_STATIC_LINK': 'CAFFE2_STATIC_LINK_CUDA',
             'CUDNN_LIB_DIR': 'CUDNN_LIBRARY'
+            'USE_GLOO_IBVERBS': 'USE_IBVERBS'   # Backward compatibility. Will be removed in the future.
```
Comment (xuhdev) [inline tools/setup_helpers/cmake.py]: @ezyang weird, why does this line appear here? It was added under additional_options

**Comment by xuhdev** [comment]
@ezyang I've redone the rebase

**Review by ezyang — APPROVED** [review]

**Comment by xuhdev** [comment]
Error doesn't seem to be related. Let me rebase again

@pytorchbot rebase this please

**Comment by xuhdev** [comment]
Finally after a streak of rebasis!

**Pull request closed without merging**